File: src/main/java/com/matchFit/common/dto/response/ApiResponseDTO.java
package com.matchFit.common.dto.response;

import com.matchFit.common.code.ErrorCode;
import com.matchFit.common.code.SuccessCode;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ApiResponseDTO<T> {
    private final String code;
    private final String message;
    private final T data;

    public static <T> ApiResponseDTO<T> onSuccess(T data) {
        return new ApiResponseDTO<>("SUCCESS", "요청이 정상 처리되었습니다.", data);
    }

    public static <T> ApiResponseDTO<T> onSuccess(SuccessCode code, T data) {
        return new ApiResponseDTO<>(code.getCode(), code.getMessage(), data);
    }

    public static <T> ApiResponseDTO<T> onFailure(ErrorCode code, T data) {
        return new ApiResponseDTO<>(code.getCode(), code.getMessage(), data);
    }
}

File: src/main/kotlin/com/matchFit/common/dto/response/ApiResponseDTO.kt
package com.matchFit.common.dto.response

import com.matchFit.common.code.ErrorCode
import com.matchFit.common.code.SuccessCode

data class ApiResponseDTO<T>(
    val code: String,
    val message: String,
    val data: T?
) {
    companion object {
        fun <T> onSuccess(data: T?): ApiResponseDTO<T> =
            ApiResponseDTO("SUCCESS", "요청이 정상 처리되었습니다.", data)

        fun <T> onSuccess(code: SuccessCode, data: T?): ApiResponseDTO<T> =
            ApiResponseDTO(code.code, code.message, data)

        fun <T> onFailure(code: ErrorCode, data: T?): ApiResponseDTO<T> =
            ApiResponseDTO(code.code, code.message, data)
    }
}
